OpenSky.API/Controllers/FinancialController.cs

```diff
@@ -133,6 +133,57 @@ public async Task<ActionResult<ApiResponse<string>>> BobsYourUncle()
             }
         }
 
+        /// -------------------------------------------------------------------------------------------------
+        /// <summary>
+        /// Resets user balance
+        /// </summary>
+        /// <remarks>
+        /// Damir, 29/11/2023
+        /// </remarks>
+        /// <returns>
+        /// An asynchronous result that yields the account balances.
+        /// </returns>
+        /// -------------------------------------------------------------------------------------------------
+        [HttpPost("resetFunds", Name = "ResetFunds")]
+        public async Task<ActionResult<ApiResponse<string>>> ResetFunds()
+        {
+            try
+            {
+                this.logger.LogInformation($"{this.User.Identity?.Name} | POST Financial/resetFunds");
+
+                // ReSharper disable AssignNullToNotNullAttribute
+                var user = await this.userManager.FindByNameAsync(this.User.Identity?.Name);
+                if (user == null)
+                {
+                    return new ApiResponse<string> { Message = "Unable to find user record!", IsError = true };
+                }
+
+                var resetRecord = new FinancialRecord
+                {
+                    ID = Guid.NewGuid(),
+                    Timestamp = DateTime.UtcNow,
+                    UserID = user.Id,
+                    Category = FinancialCategory.None,
+                    Expense = user.PersonalAccountBalance - user.PersonalAccountBalance,
+                    Description = "Funds have been reset"
+                };
+                await this.db.FinancialRecords.AddAsync(resetRecord);
+
+                var saveEx = await this.db.SaveDatabaseChangesAsync(this.logger, "Failed to reset funds");
```

Comment:
Before saving the changes you also need to actually adjust the user's balance. The financial record above is just the trace record of what happened.

user.PersonalAccountBalance = 0;